### PR TITLE
Add support for push scheduling

### PR DESCRIPTION
**Pull request by felipemobile**
Add a configuration flag on the server to handle the availability of
push scheduling.

### CODE REVIEW

File: src/Controllers/PushController.js [hunk 1]
```diff
@@ -84,6 +88,34 @@ export class PushController {
...
     return expirationTime.valueOf();
   }
+
+  /**
+   * Get push time from the request body.
+   * @param {Object} request A request object
+   * @returns {Number|undefined} The push time if it exists in the request
+   */
+  static getPushTime(body = {}) {
+    var hasPushTime = !!body['push_time'];
```
Comment (flovilmart) [inline src/Controllers/PushController.js]: use `body.hasOwnProperty('push_time')`

File: src/Controllers/PushController.js [hunk 2]
```diff
@@ -49,6 +50,9 @@ export class PushController {
       onPushStatusSaved(pushStatus.objectId);
       return badgeUpdate();
     }).then(() => {
+      if (body.push_time) {
```
Comment (flovilmart) [inline src/Controllers/PushController.js]: if body.push_time is set and the config is not enabled for scheduling, we should warn that the message will be delivered right away no? Only skip if configured, what do you think? I'm thinking of all other users :)

File: src/Controllers/PushController.js (hunk 1 above)
Comment (felipemobile) [inline src/Controllers/PushController.js]: I've used the same convention found in the getExpirationTime function.

https://github.com/felipemobile/parse-server/blob/master/src/Controllers/PushController.js#L66

Comment (flovilmart) [inline src/Controllers/PushController.js]: Yeah, bad conventions from when I started a year ago

File: src/Controllers/PushController.js (hunk 2 above)
Comment (felipemobile) [inline src/Controllers/PushController.js]: That makes sense. I've updated the code to only allow skipping on the right configuration.

Comment (felipemobile) [inline src/Controllers/PushController.js]: @flovilmart What do you mean about CLI definitions?

Comment (flovilmart) [inline src/Controllers/PushController.js]: https://github.com/parse-community/parse-server/blob/master/src/cli/definitions/parse-server.js

**Comment by felipemobile** [comment]
@flovilmart the code is updated with additional commits :)

File: src/Controllers/PushController.js [hunk 3]
```diff
@@ -84,6 +88,34 @@ export class PushController {
     }
     return expirationTime.valueOf();
   }
+
+  /**
+   * Get push time from the request body.
+   * @param {Object} request A request object
+   * @returns {Number|undefined} The push time if it exists in the request
+   */
+  static getPushTime(body = {}) {
```
Comment (flovilmart) [inline src/Controllers/PushController.js]: could we add a test for that piece?

There is a bit of logic there, even trivial but that would help not introduce any regression. 

I could add them on the top of your branch if you don't have much time :)

Comment (felipemobile) [inline src/Controllers/PushController.js]: Sure, we've to add a test. Any help is appreciated.

Comment (felipemobile) [inline src/Controllers/PushController.js]: i wish i knew enough to help.. i will check your test to help further in any future contribution. Tks in advance @flovilmart

**Comment by flovilmart** [comment]
@felipemobile thanks for getting that one started, I just merged #3722

**Pull request closed without merging**